pkg/controller/keepalivedgroup/keepalivedgroup_controller.go

```diff
@@ -236,8 +235,26 @@ func getNamespaceNameKey(obj metav1.Object) string {
 	return obj.GetNamespace() + "/" + obj.GetName()
 }
 
+func isInList(sid int, ids []int) bool {
+	for _, id := range ids {
+		if sid == id {
+			return true
+		}
+	}
+	return false
+}
+
 func assignRouterIDs(instance *redhatcopv1alpha1.KeepalivedGroup, services []corev1.Service) (bool, error) {
 	assignedServices := []string{}
+	assignedIDs := []int{}
+	if len(instance.Spec.BlacklistRouterIDs) > 0 {
+		assignedIDs = append(assignedIDs, instance.Spec.BlacklistRouterIDs...)
+		for key, val := range instance.Status.RouterIDs {
+			if isInList(val, instance.Spec.BlacklistRouterIDs) {
+				delete(instance.Status.RouterIDs, key)
```

Comment:
For consistency, could these values be added to list toBeRemovedSet (It would have to be created earlier in the function)

Reply:
No, this is the simplest and most effective solution. Your suggestion would break the strset computing logic

Reply:
Could you elaborate on that? Could you not change line 254 to `append(toBeRemovedSet,val)`, and then change 267 to `toBeRemovedSet = append(toBeRemovedSet,strset.Difference(assignedServicesSet, lbServicesSet)) ?